runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStylesPlugin.java

```diff
@@ -263,6 +301,22 @@ else if ((attackStyle == CASTING) && (castingMode == 1))
 		}
 	}
 
+	public void updateAutoRetaliate(Boolean enabled)
+	{
+		autoRetaliateVarbit = client.getVar(VarPlayer.AUTO_RETALIATE);
```

Comment:
Cant this be completely moved to onVarbitChanged event and removed from here?

Reply:
How do you mean?

Reply:
As far as I'm aware, those onChange events are only for the AttackStylesPluginTest. They don't do anything.

Reply:
This autoRetaliateVarbit is already set in varbit changed. No need for it to be set again here. Also you probably dont need warnedRelaliateSelected at all because you can just check the varbit in-place.

Reply:
They do. They are events that are published from EventBus

Reply:
Varbit changed. What line is that? Or is it another java class.

Reply:
There are multiple methods with parameter VarbitChanged in this class, that is incorrect, and you are adding 1 more incorrect thing (by adding another method with param VarbitChanged) to this mess. So when at it, you probably should just merge all 3 (or 4 now) methods into one as it is basically same event.

Reply:
I will remove the warnedRetaliateSelected and onAutoRetaliateChanged method. I will leave the original code as is.